tests/integrationtest/r/expression/cast.result

```diff
@@ -97,3 +97,10 @@ select 1 from t where y = YEAR(CURDATE());
 select cast(cast('14:15' as time) as year) = YEAR(CURDATE());
 cast(cast('14:15' as time) as year) = YEAR(CURDATE())
 1
+explain select null as a union all select 'a' as a;
```

Comment:
What is the explain result without this pr?

Reply:
```
mysql> explain select null as a union all select 'a' as a;
+------------------------+---------+------+---------------+---------------------------------------------------+
| id                     | estRows | task | access object | operator info                                     |
+------------------------+---------+------+---------------+---------------------------------------------------+
| Union_8                | 2.00    | root |               |                                                   |
| ├─Projection_10        | 1.00    | root |               | cast(from_binary(<nil>), var_string(1))->Column#3 |
| │ └─TableDual_11       | 1.00    | root |               | rows:1                                            |
| └─Projection_12        | 1.00    | root |               | a->Column#3                                       |
|   └─TableDual_13       | 1.00    | root |               | rows:1                                            |
+------------------------+---------+------+---------------+---------------------------------------------------+
5 rows in set (0.00 sec)
```